converter/shieldhit/parser.py

```diff
@@ -207,6 +214,45 @@ def _get_detector_by_uuid(self, detect_uuid: str) -> Optional[str]:
 
         raise ValueError(f"No detector with uuid {detect_uuid}")
 
+    def _parse_quantity_settings(self, quantity_dict: dict) -> dict or None:
+        """Parses settings from the input json into the quantity settings property"""
+
+        def create_name_from_settings() -> str:
+            """Create a name for the quantity from its settings."""
+            
+            # If the quantity has generic name in format [Quantity_XYZ], we want to use more descriptive name
+            # New name will be in format [Absolute/Rescaled]_[Quantity_XYZ]_[QuantityKeyword]_[to_Medium/to_Material]
+            # Specific elements of the name will be added only if they are present in the settings
+            if re.search(r'^Quantity(_\d*)?$', quantity_dict['name']):
+                prefix = None
+                suffix = None
+                if 'primaries' in quantity_dict:
+                    prefix = 'Absolute' 
+                elif 'rescale' in quantity_dict:
+                    prefix = 'Rescaled'
+                if 'medium' in quantity_dict:
+                    suffix = f'to_{quantity_dict["medium"]}'
+                elif 'materialUuid' in quantity_dict:
+                    suffix = f'to_{self._get_material_by_uuid(quantity_dict["materialUuid"]).sanitized_name}'
+                return '_'.join(filter(None, [prefix, quantity_dict['keyword'], quantity_dict['name'], suffix]))
```

Comment:
```suggestion
                prefix = ''
                suffix = ''
                if 'primaries' in quantity_dict:
                    prefix = 'Absolute_' 
                elif 'rescale' in quantity_dict:
                    prefix = 'Rescaled_'
                if 'medium' in quantity_dict:
                    suffix = f'_to_{quantity_dict["medium"]}'
                elif 'materialUuid' in quantity_dict:
                    suffix = f'_to_{self._get_material_by_uuid(quantity_dict["materialUuid"]).sanitized_name}'
                result = f"{prefix}{quantity_dict['keyword']}_{quantity_dict['name']}{suffix}"
                return result
```

The `'_'.join(filter(None, ` looks overly complex, why not composing the result from parts ?